<launch>

  <arg name="robot" default="orcvio_robot"/>
  <arg name="fixed_frame_id" default="world"/>

  <!-- bag parameters -->
  <arg name="path_bag" default="/mnt/disk1/orcvio/warthog_rosbags/gravel-path-3_2020-08-03-08-26-36.bag" />
  <!-- <arg name="path_bag" default="/mnt/disk1/orcvio/warthog_rosbags/forest-path-3_2020-08-03-09-44-27.bag" /> -->
  <arg name="visualize_rviz" default="true" />

  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/warty/forward/color/image_rect_color raw out:=/warty/forward/color/image_rect" />

  <!-- System Manager Nodelet  -->
  <group ns="$(arg robot)">
    <node pkg="nodelet" type="nodelet" name="system"
      args='standalone orcvio/SystemNodelet'
      output="screen">

      <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
      <param name="child_frame_id" value="odom"/>

      <remap from="~imu" to="/warty/forward/imu"/>
      <remap from="~cam0_image" to="/warty/forward/color/image_rect"/>

      <param name="config_file_template" value="$(find orcvio)/../../../config/warthog.yaml"/>
      <param name="config_file" value="$(find orcvio)/../../../cache/warthog_exp.yaml"/>
      <param name="result_file" value="$(find orcvio)/../../../cache/temp_rmse.txt"/>

      <param name="load_groundtruth_flag" type="int" value="0"/>
      <!-- <param name="use_left_perturbation_flag" type="int" value="1" /> -->

    </node>
  </group>

  <node name="publish_gt_path" pkg="orcvio" type="publish_gt_path" output="screen" clear_params="true"
        required="false">
      <param name="topic_pose"  type="string" value="/warty/pose" />
  </node>

    <!-- play the dataset -->
    <node pkg="rosbag" type="play" name="rosbag" args="$(arg path_bag)" required="true"/>
    
    <!-- open rviz  -->
    <group if="$(arg visualize_rviz)" >
      <node pkg="tf" type="static_transform_publisher" name="tf" args="0.0 0.0 0.0 0.0 0.0 0.0 1.0 world global 1000"/>
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find orcvio)/../../../ros_wrapper/src/orcvio/launch/rviz/orcvio_warthog.rviz" />
    </group>

</launch>